src/components/Datatable/Datatable.tsx

```diff
@@ -172,20 +176,31 @@ const Datatable = <D extends Record<string, unknown>>({
     [onRowsSelect],
   );
 
+  const memoizedTableConfig = useDeepCompareMemo(
```

Comment:
Good optimization! What is the problematic dependency, `restTableConfig`?

Reply:
It's one object, so that was the primary issue

Reply:
Yep, I wasn't sure if same handler fn also wasn't causing unnecessary call